### PR TITLE
feat(clickhouse): Fix roundtrips of DATE/TIMESTAMP functions

### LINKED ISSUE #3679: Makes upper case last arg in Clickhouse date_sub func
```python
from sqlglot import parse_one
from sqlglot.dialects.clickhouse import ClickHouse
tree = parse_one("SELECT date_sub(second , seconds_in_status, created) FROM table ", read=ClickHouse)

print(tree.sql(dialect=ClickHouse))
```
got: > `SELECT date_sub(second, seconds_in_status, CREATED) FROM table`
want: > `SELECT date_sub(second, seconds_in_status, created) FROM table`
or want: > `SELECT date_sub(SECOND, seconds_in_status, created) FROM table`

On exec sql in our clickhouse getting unknown column error 
Clickhouse docs: https://clickhouse.com/docs/en/sql-reference/functions/date-time-functions#date_sub

### PR DESCRIPTION
**Pull request by VaggelisD**
Fixes #3679

This PR aims to fix the roundtrip for the following CH datetime functions and their aliases:

`DATE_SUB`, `DATE_ADD`, `DATE_DIFF`, `TIMESTAMP_ADD`, `TIMESTAMP_SUB`

These functions (except `DATE_DIFF`) support either one or both of the following versions

- 2-arg based version `<func>(date, INTERVAL unit)`, supported by `DATE_ADD`, `DATE_SUB`, `TIMESTAMP_SUB`
- 3-arg based version `<func>(unit, value, date)`, supported by `DATE_ADD`, `DATE_SUB`, `TIMESTAMP_ADD`

The exception to that is `DATE_DIFF` which parses into `date_diff('unit', startdate, enddate, [timezone])`; Note that support for `timezone` was not added in this PR.

### CODE REVIEW

File: sqlglot/dialects/clickhouse.py [hunk 1]
```diff
@@ -17,10 +17,17 @@
...
     timestamptrunc_sql,
+    unit_to_var,
 )
+from sqlglot.generator import Generator
 from sqlglot.helper import is_int, seq_get
 from sqlglot.tokens import Token, TokenType
 
+DATEΤΙΜΕ_DELTA = t.Union[exp.DateAdd, exp.DateDiff, exp.DateSub, exp.TimestampSub, exp.TimestampAdd]
+
+if t.TYPE_CHECKING:
```
Comment (tobymao) [inline sqlglot/dialects/clickhouse.py:28]: what is this for? delete it

File: tests/dialects/test_clickhouse.py [hunk 2]
```diff
@@ -880,3 +880,18 @@ def test_drop_on_cluster(self):
         for creatable in ("DATABASE", "TABLE", "VIEW", "DICTIONARY", "FUNCTION"):
             with self.subTest(f"Test DROP {creatable} ON CLUSTER"):
                 self.validate_identity(f"DROP {creatable} test ON CLUSTER test_cluster")
+
+    def test_datetime_funcs(self):
+        # Function aliases are roundtripped back to one name e.g. DATE_SUB(), DATESUB() -> DATE_SUB()
+        func_names = (("DATE_SUB", "DATESUB"), ("DATE_ADD", "DATEADD"))
+        for funcs in func_names:
+            # Functions of type <func>(date, unit)
+            for func in (*funcs, ("TIMESTAMP_ADD", "TIMESTAMPADD")):
```
Comment (georgesittas) [inline tests/dialects/test_clickhouse.py:889]: This is a bit awkward, not sure if it's written correctly: `*funcs` expands to each element in the tuples (so you get back `DATE_SUB`, DATESUB` and then `("TIMESTAMP_ADD", "TIMESTAMPADD")` in the same iteration.

Can we clean these tests up a bit? They're hard to follow

**Review by georgesittas — APPROVED** [review]
One comment re: testing, LGTM otherwise

File: tests/dialects/test_clickhouse.py (hunk 2 above)
Comment (VaggelisD) [reply tests/dialects/test_clickhouse.py]: Fixed it and made it more readable, let me know if it's better

**Review by georgesittas — APPROVED** [review]
